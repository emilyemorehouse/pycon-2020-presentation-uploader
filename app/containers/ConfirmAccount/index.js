export { default as ConfirmAccount } from './ConfirmAccount'
