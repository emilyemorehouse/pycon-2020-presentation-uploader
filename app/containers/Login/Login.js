/* @flow */
import React from 'react'
import { isEmail } from 'validator'

// Components
import { Anchor } from 'components/Anchor'
import { Box } from 'components/Box'
import { Button } from 'components/Button'
import { Form, FormField } from 'components/Form'
import { Heading } from 'components/Heading'
import { PasswordInput } from 'components/PasswordInput'
import { TextInput } from 'components/TextInput'

/**
 *
 * Login
 *
 */
const Login = () => (
  <>
    <Box align="center" pad="large">
      <Heading level="2">Login</Heading>
      <Anchor href="/register" label="or Create An Account"></Anchor>
    </Box>

    <Form
      validate="blur"
      onSubmit={({ value }) => {
        // eslint-disable-next-line no-console
        console.log('Submit', value)
      }}
    >
      <FormField
        label="Email"
        name="email"
        required
        validate={[
          email => {
            if (email && !isEmail(email)) return 'Please enter a valid email address'
            return undefined
          },
        ]}
      >
        <TextInput name="email" type="email" />
      </FormField>

      <FormField
        label="Password"
        name="password"
        required
        validate={[
          password => {
            if (password && password.length <= 8) return 'Please enter more than 8 characters'
            return undefined
          },
        ]}
      >
        <PasswordInput id="password" name="password" />
      </FormField>

      {/* Submit */}
      <Box direction="row" margin={{ top: 'medium' }}>
        <Button type="submit" label="Login" primary />
      </Box>
    </Form>
  </>
)

export default Login
