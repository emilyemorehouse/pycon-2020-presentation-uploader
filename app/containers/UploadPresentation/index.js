export { default as UploadPresentation } from './UploadPresentation'
