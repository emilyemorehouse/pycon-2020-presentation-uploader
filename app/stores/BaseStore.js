import { forEach } from 'lodash'
import { action } from 'mobx'

export class BaseStore {
  /**
   * mobx-sync does not set the default values without interacting with at
   * least one of the class variables that are set up as an observable
   *
   * This takes the intial state provided to the class instantiation and runs
   * through each value from the `initialState` that must be set to ensure the
   * default values are correct
   *
   * @param {object} initialState
   */
  constructor(initialState = {}) {
    forEach(initialState, (value, key) => {
      this[key] = value
    })
    this._initialState = initialState
  }

  @action.bound
  reset() {
    forEach(this._initialState, (value, key) => {
      this[key] = value
    })
  }
}
