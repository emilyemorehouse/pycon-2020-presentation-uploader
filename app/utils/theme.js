import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Roboto Mono',
      size: '16px',
    },
    colors: {
      brand: colors.primary,
      focus: colors.secondary,
    },
  },
  button: {
    border: {
      radius: '4px',
    },
  },
}

export default theme
