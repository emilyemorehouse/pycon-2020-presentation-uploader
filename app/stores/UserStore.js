import { action, computed, observable } from 'mobx'
import { ignore } from 'mobx-sync'
import { createContext } from 'react'

import { BaseStore } from './BaseStore'

export const INITIAL_STATE = {
  accessToken: '',
  error: '',
  isLoading: false,
  message: '',
  refreshToken: '',
  user: {},
}

export class UserStore extends BaseStore {
  @observable accessToken = INITIAL_STATE.accessToken

  @ignore
  @observable
  error = INITIAL_STATE.error

  @ignore
  @observable
  isLoading = INITIAL_STATE.isLoading

  @ignore
  @observable
  message = INITIAL_STATE.message

  @observable refreshToken = INITIAL_STATE.refreshToken

  @observable user = INITIAL_STATE.user

  @computed
  get isAuthenticated() {
    return this.accessToken !== ''
  }

  @action.bound
  setCurrentUser({ accessToken = '', refreshToken = '', ...user } = {}) {
    this.accessToken = accessToken
    this.refreshToken = refreshToken
    this.user = {
      ...user,
    }
  }
}

export const user = new UserStore(INITIAL_STATE)
export const UserStoreContext = createContext(user)
