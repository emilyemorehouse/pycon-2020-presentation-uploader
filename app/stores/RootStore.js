/** @todo Fix import cycle */
/* eslint-disable import/no-cycle */

import { ignore } from 'mobx-sync'
import { action } from 'mobx'
import { createContext } from 'react'

import { locale } from './LocaleStore'
import { user } from './UserStore'
import { refreshAccessToken as refreshAccessTokenService } from '../services/user.service'

export class RootStore {
  @ignore storeLoaded = false

  locale = locale

  user = user

  @action.bound
  clearStore() {
    this.user.reset()
  }

  @action.bound
  async triggerRefreshToken(failedRequest) {
    try {
      const { access } = await refreshAccessTokenService(this.refreshToken)
      this.user.accessToken = access

      // Reassign to `failedRequest` as this object is used by the token
      // interceptor to retry the request
      // eslint-disable-next-line no-param-reassign
      failedRequest.response.config.headers.Authorization = `Bearer ${access}`
    } catch (err) {
      this.clearStore()
      throw err
    }
  }
}

export const store = new RootStore()
export const RootStoreContext = createContext(store)
