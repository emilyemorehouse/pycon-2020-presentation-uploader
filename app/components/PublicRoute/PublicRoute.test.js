import React from 'react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'
import { render, waitForElement } from '@testing-library/react'
import { Grommet } from 'grommet'
import '@testing-library/jest-dom/extend-expect'

// Components
import { Login } from '../../containers/Login'
import PublicRoute from './PublicRoute'

// Helpers
import { DEFAULT_LOCALE } from '../../i18n'
import theme from '../../utils/theme'

// Stores
import trunk from '../../configureStore'
import { UserStore, UserStoreContext } from '../../stores/UserStore'

describe('<PublicRoute />', () => {
  let userStore = null

  beforeAll(() => {
    userStore = new UserStore()
    trunk.init()
  })

  const renderComponent = store =>
    render(
      <Grommet theme={theme}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter initialEntries={['/login']}>
            <UserStoreContext.Provider value={store}>
              <PublicRoute path="/login" component={Login} />
            </UserStoreContext.Provider>
          </MemoryRouter>
        </IntlProvider>
      </Grommet>,
    )

  it('renders Login if user is not authenticated', async () => {
    const { container, getByTestId } = renderComponent(userStore)
    expect(container).toMatchSnapshot()
    const LoginPage = await waitForElement(() => getByTestId('login-header'))
    expect(LoginPage).toBeVisible()
  })

  it('does not render Login if user is authenticated', async () => {
    userStore = {
      accessToken: 'faketoken',
    }
    const { container } = renderComponent(userStore)
    expect(container).toMatchSnapshot()
  })
})
