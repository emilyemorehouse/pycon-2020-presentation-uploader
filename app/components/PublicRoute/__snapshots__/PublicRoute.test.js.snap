// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicRoute /> does not render Login if user is authenticated 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 blnwXx"
  />
</div>
`;

exports[`<PublicRoute /> renders Login if user is not authenticated 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 blnwXx"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 DKkMM"
      data-testid="login-header"
    >
      <h2
        class="StyledHeading-sc-1rdh4aw-0 jMKEAN"
      >
        Login
      </h2>
      <a
        class="StyledAnchor-sc-1rp7lwl-0 kZUeBZ"
        href="/register"
      >
        or Create An Account
      </a>
    </div>
    <form
      id="login-form"
      style="max-width: 70vw; margin: auto;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 euhBMS"
      >
        <label
          class="StyledText-sc-1sadyjn-0 enbIRN"
        >
          Email*
        </label>
        <div
          class="StyledBox-sc-13pk1d4-0 GCqjV"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 jJdKAu"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 bjXYsX"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 euhBMS"
      >
        <label
          class="StyledText-sc-1sadyjn-0 enbIRN"
        >
          Password*
        </label>
        <div
          class="StyledBox-sc-13pk1d4-0 GCqjV"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 jJdKAu"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 emmXTg"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 bjXYsX"
                  id="password"
                  name="password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="StyledButton-sc-323bzc-0 jvkqEN Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
                type="button"
              >
                <svg
                  aria-label="Hide"
                  class="StyledIcon-ofa7kd-0 gzViXn"
                  style="width: 30px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 gVZJeo"
      >
        <button
          class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
